src/coreclr/hosts/corerun/corerun.cpp

```diff
@@ -1,544 +1,322 @@
+    string_t clr_path;
 
-            // Check for %CORE_ROOT% and try to load CoreCLR.dll from it if it is set
-            StackSString coreRoot;
-            m_coreCLRModule = NULL; // Initialize this here since we don't call TryLoadCoreCLR if CORE_ROOT is unset.
-            if (WszGetEnvironmentVariable(W("CORE_ROOT"), coreRoot) > 0 && coreRoot.GetCount() > 0)
-            {
-                coreRoot.Append(W('\\'));
-                m_coreCLRModule = TryLoadCoreCLR(coreRoot);
-            }
-            else
-            {
-                *m_log << W("CORE_ROOT not set; skipping") << Logger::endl;
-                *m_log << W("You can set the environment variable CORE_ROOT to point to the path") << Logger::endl;
-                *m_log << W("where CoreCLR.dll lives to help CoreRun.exe find it.") << Logger::endl;
-            }
+    // The full path to the Supplied managed entry assembly.
+    string_t entry_assembly_fullpath;
 
-            // Try to load CoreCLR from the directory that coreRun is in
-            if (!m_coreCLRModule)
-            {
-                m_coreCLRModule = TryLoadCoreCLR(m_hostDirectoryPath);
-            }
+    // Arguments to pass to managed entry assembly.
+    int entry_assembly_argc;
+    const char_t** entry_assembly_argv;
 
-            if (!m_coreCLRModule)
-            {
+    // Wait for debugger to be attached.
+    bool wait_to_debug;
 
-                // Failed to load. Try to load from the well-known location.
-                wchar_t coreCLRInstallPath[MAX_LONGPATH];
-                ::ExpandEnvironmentStringsW(coreCLRInstallDirectory, coreCLRInstallPath, MAX_LONGPATH);
-                m_coreCLRModule = TryLoadCoreCLR(coreCLRInstallPath);
+    // Perform self test.
+    bool self_test;
+};
 
-            }
+namespace envvar
+{
+    // Points to a path containing the CoreCLR binary.
+    const char_t* coreRoot = W("CORE_ROOT");
 
-            if (m_coreCLRModule)
-            {
+    // Points to a path containing additional platform assemblies.
+    const char_t* coreLibraries = W("CORE_LIBRARIES");
 
-                // Save the directory that CoreCLR was found in
-                DWORD modulePathLength = WszGetModuleFileName(m_coreCLRModule, m_coreCLRDirectoryPath);
+    // Variable used to preload a mock hostpolicy for testing.
+    const char_t* mockHostPolicy = W("MOCK_HOSTPOLICY");
 
-                // Search for the last backslash and terminate it there to keep just the directory path with trailing slash
-                SString::Iterator lastBackslash = m_coreCLRDirectoryPath.End();
-                m_coreCLRDirectoryPath.FindBack(lastBackslash, W('\\'));
-                m_coreCLRDirectoryPath.Truncate(lastBackslash + 1);
+    // Name of the environment variable controlling server GC.
+    // If set to 1, server GC is enabled on startup. If 0, server GC is
+    // disabled. Server GC is off by default.
+    const char_t* serverGc = W("COMPlus_gcServer");
```

Comment:
I believe `COMPlus_gcServer`  and `COMPlus_gcConcurrent` are also read and respected by the runtime. No sure why we have it duplicated here.

Reply:
I believe because of https://github.com/dotnet/runtime/blob/61c69d157a008bcee89adc234f0b2822087fc253/src/coreclr/dlls/mscoree/unixinterface.cpp#L95-L115

Reply:
If `COMPlus_gcServer` is not read by the runtime, it would be nice to delete it from `src\coreclr\inc\clrconfigvalues.h`

Reply:
Okay. I am going to remove the globalization invariant first and then merge this in. These other values are going to require more inspection.

Reply:
@jkotas Both the `COMPlus_gcServer` and `COMPlus_gcConcurrent` are read by the runtime. It is done first in the call to `Configuration::GetKnobBooleanValue()`. Do you have a preference here? I think I am of the opinion that special casing these in `corerun` is just adding unnecessary complexity. I vote to remove them. @vitek-karas and @janvorli for their opinions as well.

Reply:
> I think I am of the opinion that special casing these in corerun is just adding unnecessary complexity. I vote to remove them

I agree that we should remove the special casing in corerun.

Reply:
See https://github.com/dotnet/runtime/pull/49950